### PR TITLE
Fix violations that occur with new analyzers

### PR DESCRIPTION
**Pull request by pakdev**
# Justification
Updating to the latest analyzers in #67 revealed some violations that can/should be solved in the current codebase first.

# Implementation
Solved most of the violations that the latest analyzers complained about. Those that remain require newer APIs.

# Testing
Ensured build/tests still pass.

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -44,21 +44,20 @@
 
     <PropertyGroup>
         <IncludeCompatabilityAnalyzer Condition=" '$(IncludeCompatabilityAnalyzer)' == '' ">True</IncludeCompatabilityAnalyzer>
+        <RunAnalyzersDuringBuild>true</RunAnalyzersDuringBuild>
+        <RunAnalyzersDuringLiveAnalysis>true</RunAnalyzersDuringLiveAnalysis>
+        <RunAnalyzers>true</RunAnalyzers>
```
Comment (jryckman) [inline Directory.Build.props]: Did we have IncludeCompatibilityAnalyzer set to true before? I'm trying to understand this particular change and why you moved it here

**Review by jryckman — APPROVED** [review]
I left one main comment about the move in the property group.

File: Directory.Build.props (hunk 1 above)
Comment (pakdev) [reply Directory.Build.props]: Yes; I simply moved these 3 properties into the IncludeCompatbilityAnalyzer's `PropertyGroup`, but I could put them into their own if desired for readability

Comment (pakdev) [reply Directory.Build.props]: I moved them out of the `PropertyGroup` that only evaluates during debug

Comment (pakdev) [reply Directory.Build.props]: I just killed them. Apparently they all default to `true`

Comment (jryckman) [reply Directory.Build.props]: Sounds good

**Pull request merged**